src/XMakeBuildEngine/UnitTests/BackEnd/IntrinsicTask_Tests.cs

```diff
@@ -2098,15 +2098,15 @@ public void RemoveWithWildcards()
                 <Target Name='t'>
                     <ItemGroup>
                         <i1 Include='" + files.First() + ";" + files.ElementAt(1) + @";other'/> 
-                        <i1 Remove='$(temp)" + Path.DirectorySeparatorChar + @"*.tmp'/>
+                        <i1 Remove='$(temp)*.tmp'/>
```

Comment:
Mkay, so this is how msbuild behaves now on windows. I tried out dev14 and our xplat branch.

```xml
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
        <ItemGroup>
                <I Include="a//b//**" Exclude="a/b/*.foo"/>
                <I2 Include="a/b/**" Exclude="a//b/*.foo"/>
        </ItemGroup>

        <Target Name="Build">
                <ItemGroup>
                        <T Include="a/b/**"/>
                        <T Remove="a//b/*.foo"/>
                        <T Remove="a/b//*.foo"/>
                </ItemGroup>
                <Message Text="I:@(I)" Importance="High"/>
                <Message Text="I2:@(I2)" Importance="High"/>
                <Message Text="T:@(T)" Importance="High"/>
        </Target>
</Project>
```

```
D:\projects\tests\projects\play> msbuild
Microsoft (R) Build Engine version 14.0.25420.1
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2/2/2017 11:41:05 AM.
Project "D:\projects\tests\projects\play\build.proj" on node 1 (default targets).
Build:
  I:a//b//a.cs;a//b//b.foo
  I2:a/b/a.cs;a/b/b.foo
  T:a/b/a.cs;a/b/b.foo
Done Building Project "D:\projects\tests\projects\play\build.proj" (default targets).


Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.02
D:\projects\tests\projects\play> D:\projects\msbuild\bin\Bootstrap\15.0\Bin\MSBuild.exe
Microsoft (R) Build Engine version 15.1.543.24285
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2/2/2017 11:41:12 AM.
Project "D:\projects\tests\projects\play\build.proj" on node 1 (default targets).
Build:
  I:a//b//a.cs;a//b//b.foo
  I2:a/b/a.cs;a/b/b.foo
  T:a/b/a.cs;a/b/b.foo
Done Building Project "D:\projects\tests\projects\play\build.proj" (default targets).


Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.35
D:\projects\tests\projects\play> D:\projects\msbuild\bin\Bootstrap-NetCore\CoreRun.exe D:\projects\msbuild\bin\Bootstrap
-NetCore\MSBuild.dll
Microsoft (R) Build Engine version 15.1.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2/2/2017 11:41:33 AM.
Project "D:\projects\tests\projects\play\build.proj" on node 1 (default targets).
Build:
  I:a//b//a.cs;a//b//b.foo
  I2:a/b/a.cs;a/b/b.foo
  T:a/b/a.cs;a/b/b.foo
Done Building Project "D:\projects\tests\projects\play\build.proj" (default targets).
```

So for Windows at least the behaviour is the same.

On non-windows xplat msbuild gives you this:

```
Microsoft (R) Build Engine version 15.2.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2/2/17 12:06:30 PM.
Project "/Users/micodoba/Dropbox/build.proj" on node 1 (default targets).
Build:
  I:a/b/a.cs
  I2:a/b/a.cs
  T:a/b/a.cs
Done Building Project "/Users/micodoba/Dropbox/build.proj" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.74
```

And with this PR it would give you this:

```
Microsoft (R) Build Engine version 15.1.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2/2/17 12:09:37 PM.
Project "/Users/micodoba/Dropbox/build.proj" on node 1 (default targets).
Build:
  I:a//b//a.cs;a//b//b.foo
  I2:a/b/a.cs;a/b/b.foo
  T:a/b/a.cs;a/b/b.foo
Done Building Project "/Users/micodoba/Dropbox/build.proj" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.81
```